src/sentry/reprocessing2.py

```diff
@@ -347,7 +346,75 @@ def _get_info_reprocessed_key(group_id):
     return f"re2:info:{group_id}"
 
 
-def mark_event_reprocessed(data=None, group_id=None, project_id=None):
+def buffered_handle_remaining_events(
+    project_id,
+    old_group_id,
+    new_group_id,
+    datetime_to_event,
+    remaining_events,
+    force_flush_batch=False,
+):
+    """
+    A quick-and-dirty wrapper around `handle_remaining_events` that batches up
+    event IDs in Redis. We need this because Snuba cannot handle many tiny messages and prefers big ones instead.
+
+    Best performance is achieved when timestamps are close together. Luckily we
+    happen to iterate through events ordered by timestamp.
+
+    Ideally we'd have batching implemented via a service like buffers, but for
+    more than counters.
+    """
+
+    client = _get_sync_redis_client()
+    key = f"re2:remaining:{project_id}:{old_group_id}"
+
+    if datetime_to_event:
+        client.lpush(
+            key,
+            *(f"{to_timestamp(datetime)};{event_id}" for datetime, event_id in datetime_to_event),
+        )
+        client.expire(key, settings.SENTRY_REPROCESSING_SYNC_TTL)
+
+    if (
+        force_flush_batch
+        or client.llen(key) > settings.SENTRY_REPROCESSING_REMAINING_EVENTS_BUF_SIZE
+    ):
+        event_ids_batch = []
+        min_datetime = None
+        max_datetime = None
+
+        # TODO: Redis 6 introduces LPOP with <count> argument, use here?
+        while True:
+            row = client.lpop(key)
```

Comment:
Since you intend to empty the key, can you check if you can atomically take the entire key value here instead of popping them individually?

Reply:
There actually seems no way to do that.. redis doesn't have a "pop key" command. I however moved it into the other Celery task now.